flight/Libraries/timeutils.c

```diff
@@ -0,0 +1,99 @@
+/**
+ ******************************************************************************
+ * @addtogroup TauLabsLibraries Tau Labs Libraries
+ * @{
+ *
+ * @file       timeutils.c
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2015
+ * @brief      Time conversion functions
+ *
+ * @see        The GNU Public License (GPL) Version 3
+ *
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "timeutils.h"
+
+/**
+ * Convert UNIX timestamp to date/time
+ * Based on: http://stackoverflow.com/questions/21593692/convert-unix-timestamp-to-date-without-system-libs
+ * NOTE: May not handle leap-seconds correctly
+ */
+void date_from_timestamp(uint32_t timestamp, DateTimeT *date_time)
+{
+	uint32_t minutes, hours;
+	uint16_t days, year, dayOfWeek;
+	uint8_t month;
+
+	/* calculate minutes */
+	minutes  = timestamp / 60;
+	timestamp -= minutes * 60;
+	/* calculate hours */
+	hours    = minutes / 60;
+	minutes -= hours   * 60;
+	/* calculate days */
+	days     = hours   / 24;
+	hours   -= days    * 24;
+
+	/* Unix time starts in 1970 on a Thursday */
+	year      = 1970;
+	dayOfWeek = 4;
+
+	while(1)
+	{
```

Comment:
Time code makes me nervous because it's so easy to get subtly wrong.  This appears correct, but I'm not 100% confident.  This is also slow, but OK.

For reference the glibc version is at: https://sourceware.org/git/?p=glibc.git;a=blob;f=time/offtime.c;h=a437266e95eaaf7bb99e1b4512a633f9dfd906d4;hb=6b142b3a1d007d7e6f50c26710de7177bc4aca74 

It runs in constant-ish time so maybe we want to bring this in instead.